### DISCUSSION TITLE
FileUtil.createMissingParentDirectories always returns false

### DISCUSSION THREAD

**Discussion by renegat59** [qos-ch/logback#622] [Q&A]
Hi,
I am using LogBack in the company I work for and we encountered an issue with it. The problem is that in [line 58](https://github.com/qos-ch/logback/blob/master/logback-core/src/main/java/ch/qos/logback/core/util/FileUtil.java#L58) of `FileUtil` there is a call to `parent.exists()`, but that returns false if the path is relative. I didn't arrive to the source of the problem, I was trying to set `-Duser.dir` as is suggested [here](https://stackoverflow.com/questions/919918/file-exists-returns-false-when-file-exists) but it doesn't help. However I see that something like this works:
```
parent.getAbsoluteFile().exists();
```

in the image you can see the variables from my debugger:
![image](https://user-images.githubusercontent.com/2337858/209136264-bd2c7f91-e3bc-443a-a9a8-91af90a40f50.png)

As you can see the `user.dir` is correct but still the `exists()` on relative file is failing.

What do you think about changing that to use `getAbsoluteFile()`? Do you see anything against? Or maybe you have an idea how to fix it on our side? I'm happy to make this change and prepare PR if needed.

**Comment by ceki** [comment]
Can you specify the OS, JDK version where this issue occurs? 

A test case would be most appreciated. The test case can omit using logback, and involve only the `File.exists()` method.

**Comment by renegat59** [comment]
@ceki thanks for your answer. I did more tests and I realized that it was Gradle messing around with me. We have multiple modules inside one repo and running gradle from the root of the repo. Some commands had different cwd (of the module) and that's why it was failing. I managed to fix it on our end.
Thanks again!
